src/MainMenuController.cpp

```diff
@@ -263,79 +248,20 @@ void MainMenuController::animateTitleColor()
 /**
  *
  */
-void MainMenuController::updateSelectorPosition(utils::Context& context)
-{
-    /* limit de movement of the selector, according to the first and last
-       menu list item position */
-    /* TODO: #436 should be refactored in a middleware */
-    selectorPosition = (
-                           (selectorPosition > ITEM_EXIT) ?
-                           ITEM_EXIT :
-                           selectorPosition
-                       );
-
-    selectorPosition = (
-                           (selectorPosition < ITEM_NEW_GAME) ?
-                           ITEM_NEW_GAME :
-                           selectorPosition
-                       );
-
-    /* before applying the selector color on the selected menu item, we
-       first change the color of every menu item to white */
-    itemNewGame.setColor(colors::ColorsManager::get().getColorWhite());
-    itemLoadGame.setColor(colors::ColorsManager::get().getColorWhite());
-    itemEditor.setColor(colors::ColorsManager::get().getColorWhite());
-    itemOptions.setColor(colors::ColorsManager::get().getColorWhite());
-    itemExit.setColor(colors::ColorsManager::get().getColorWhite());
-
-    /* apply the selector color on the right item, according to the current
-       selector position */
-    switch(selectorPosition)
-    {
-    case ITEM_NEW_GAME:
-    {
-        itemNewGame.setColor(colorSelector);
-        break;
-    }
-    case ITEM_LOAD_GAME:
-    {
-        itemLoadGame.setColor(colorSelector);
-        break;
-    }
-    case ITEM_EDITOR:
-    {
-        itemEditor.setColor(colorSelector);
-        break;
-    }
-    case ITEM_OPTIONS:
-    {
-        itemOptions.setColor(colorSelector);
-        break;
-    }
-    default:
-    {
-        itemExit.setColor(colorSelector);
-        break;
-    }
-    }
-}
-
-/**
- *
- */
 void MainMenuController::selectMenuItem()
 {
     /* update the expected controller id according to the selector
        current position */
-    switch(selectorPosition)
+    /* TODO: the numbers are not explicit enough */
+    switch(getSelectorPosition())
     {
-    case ITEM_NEW_GAME:
+    case 0:
     {
         expectedControllerId = NEW_GAME_CONTROLLER_ID;
 
         break;
     }
-    case ITEM_EDITOR:
+    case 2:
```

Comment:
add a comment before each number to explicitly indicate which item is selected and which controller is called